### PR TITLE
[doc]metrics

### PR DESCRIPTION
**Pull request by Amber1990Zhang**
1. update storage metrics, add prefix
2. add meta and graph metrics
2. update the newly added docs in the TOC

### CODE REVIEW

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+_error_qps
+```
+
+将接口名和相应指标连接在一起即可获得完整的指标名，例如 `graph_insertVertex_latency`、`graph_insertVertex_qps`、`graph_insertVertex_error_qps`、分别代表插入一个点的延迟、QPS 和发生错误的 QPS。
+
+### 统计类型
+
+目前支持的统计类型有 SUM、COUNT、AVG、RATE 和 P 分位数 (P99，P999， ... ，P999999)。其中：
+
+- `_latency` 和 `_error_qps` 这两类后缀的指标，支持 SUM、COUNT、AVG、RATE，但不支持 P 分位；
```
Comment (critical27) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: `_latency` support P99 etc, but `_error_qps` and `_qps` does not. You'd better check all doc

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+meta_heartbeat_qps.avg.60         // 最近一分钟心跳的平均 QPS
+meta_heartbeat_error_qps.count.60   // 最近一分钟心跳的平均错误总计数量
+meta_heartbeat_latency.avg.60     // 最近一分钟心中的平均延时
+```
+
+假设本地启动了一个 nebula meta service，同时启动时设置的 `ws_http_port` 端口号为 11000。通过 HTTP 的 GET 接口发送，方法名为 get_stats，参数为 stats 加对应的指标名字。下面是通过 HTTP 接口获取指标的示例：
+
+```bash
+# 获取一个指标
+curl -G "http://127.0.0.1:12000/get_stats?stats=meta_heartbeat_qps.avg.60"
```
Comment (critical27) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md]: 12000 -> 11000 will be better?

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+每个指标名都由接口名加指标名构成，目前支持获取如下接口：
+
+```text
+graph_storageClient: 通过 storageClient 发送的请求，需要同时向多个 storage 并发多条消息时，按一次统计
+graph_metaClient: 通过 metaClient 发送的请求
+graph_graph_all: 客户端向 graph 发送的请求，当一条请求包含多条语句时，按一条计算
+graph_insertVertex: 插入点
+graph_insertEdge: 插入边
+graph_deleteVertex: 删除点
+graph_deleteEdge: 删除边  //代码还没加
```
Comment (laura-ding) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: //未支持

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+目前，**Nebula Graph** 支持通过 HTTP 方式来获取 Graph Service 层的基本性能指标。
+
+每一个性能指标都由三部分组成，分别为指标名、统计类型、时间范围。
+
+| counter\_name | statistic\_type | time_range |
+| ----  |  ----|-------|
+
+### 指标名
+
+每个指标名都由接口名加指标名构成，目前支持获取如下接口：
```
Comment (laura-ding) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: 服务名加模块名

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+### 时间范围
+
+时间范围目前只支持三种，分别为 60、600、3600，分别表示最近一分钟，最近十分钟和最近一小时。
+
+## 通过 HTTP 接口获取相应的性能指标
+
+根据上面的介绍，就可以写出一个完整的指标名称了，下面是一些示例：
+
+```cpp
+graph_insertVertex_latency.avg.60        // 最近一分钟插入一个点的平均延时
```
Comment (laura-ding) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: 最近一分钟所有插入语句执行成功的平均延时

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+graph_updateEdge_error_qps.count.3600   // 最近一小时更新一条边发生错误的总计数量
```
Comment (laura-ding) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: 最近一小时更新边命令失败的总计数量

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+graph_deleteEdge: 删除边  //代码还没加
+graph_updateVertex: 更新点的属性
+graph_updateEdge: 更新边的属性
+graph_go: 执行 go 命令
+graph_findPath: 查找最小路径或者全路径
+graph_fetchVertex: 获取点属性，不统计获取点的总数，只统计执行命令的数量
+graph_fetchEdge: 获取边属性，不统计边的总数，只统计执行命令的数量
+```
+
+每一个接口都有三个性能指标，分别为延迟(单位为 us)、QPS、发生错误的 QPS，后缀名如下：
```
Comment (laura-ding) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: 成功的QPS

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+目前，**Nebula Graph** 支持通过 HTTP 方式来获取 Meta Service 层的基本性能指标。
+
+每一个性能指标都由三部分组成，分别为指标名、统计类型、时间范围。
+
+| counter\_name | statistic\_type | time_range |
+| ----  |  ----|-------|
+
+### 指标名
+
+每个指标名都由接口名加指标名构成，meta 只统计心跳信息，目前支持获取如下接口：
```
Comment (laura-ding) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md]: ditto

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/storage-metrics.md [hunk 9]
```diff
@@ -13,19 +13,19 @@
 
 ### 指标名
 
-每个指标名都由接口名加指标名构成，目前支持获取如下接口
+每个指标名都由接口名加指标名构成，目前支持获取如下接口：
```
Comment (laura-ding) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/storage-metrics.md]: ditto

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/storage-metrics.md [hunk 10]
```diff
@@ -13,19 +13,19 @@
...
-获取边的属性 edge_props
-插入一个点 add_vertex
-插入一条边 add_edge
-删除一个点 del_vertex
-更新一个点的属性 update_vertex
-更新一条边的属性 update_edge
-读取一个键值对 get_kv
-写入一个键值对 put_kv
-仅限内部使用 get_bound
+获取点的属性 storage_vertex_props
```
Comment (laura-ding) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/storage-metrics.md]: 这里是中文说明先，graph那边是反过来，麻烦统一一下

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+- metrics have suffixes `_qps` and `_error_qps` support SUM, COUNT, AVG, RATE but don't support P quantiles.
+- metrics have suffixes `_latency` support SUM, COUNT, AVG, RATE, and P quantiles.
+
+### Time Range
+
+Currently supported time ranges are 60, 600, and 3600, which mean the last minute, the last ten minutes, and the last hour till now.
+
+## Obtain the Corresponding Metrics via HTTP Interface
+
+According to the above introduction, you can make a complete metrics name, here are some examples:
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: you can make a complete metrics name, here are some examples:

you can make a complete metrics name. Here are some examples:

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+```cpp
+graph_insertVertex_latency.avg.60   // the average latency of successfully inserting a vertex in the last minute
+graph_updateEdge_error_qps.count.3600  // total number of failures in updating an edge in the last hour
+```
+
+Assume that a graph service is started locally, and the `ws_http_port` port number is set to 13000 when starting. It is sent through the GET interface of HTTP. The method name is get_stats, and the parameter is stats plus the corresponding metrics name. Here's an example of getting metrics via the HTTP interface:
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: It is recommended to put the interface name and the method name, such as GET and get_stats in bold.

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
+# Meta Metrics
+
+## Introduction
+
+Currently, **Nebula Graph** supports obtaining the basic performance metrics for the meta service via HTTP.
+
+Each performance metrics consists of three parts, namely `<counter_name>.<statistic_type>.<time_range>`.
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md]: metrics -> metric

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+
+### Counter Names
+
+Each counter name is composed of the interface name and the counter name. Meta service only counts the heart beat. Currently, the supported interfaces are:
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md]: heart beat -> heartbeat

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+meta_heartbeat_qps
+meta_heartbeat_error_qps
+meta_heartbeat_latency
+```
+
+### Statistics Type
+
+Currently supported types are SUM, COUNT, AVG, RATE and P quantiles (P99, P999, ..., P999999). Among which:
+
+- metrics have suffixes `_qps` and `_error_qps` support SUM, COUNT, AVG, RATE but don't support P quantiles.
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md]: metrics -> Metrics

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+- metrics have suffixes `_latency` support SUM, COUNT, AVG, RATE, and P quantiles.
+
+### Time Range
+
+Currently supported time ranges are 60, 600, and 3600, which mean the last minute, the last ten minutes, and the last hour till now.
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/meta-service-administration/meta-metrics.md]: Currently, the supported time ranges are 60s, 600s, and 3600s, which correspond to the last minute, the last ten minutes, and the last hour till now.

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md (hunk 11 above)
Comment (Amber1990Zhang) [reply docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/graph-service-administration/graph-metrics.md]: Good catch.

**Pull request merged**